addon/services/current-user.js

```diff
@@ -63,4 +63,19 @@ export default Ember.Service.extend({
             promise: this.load().catch(() => null),
         });
     }),
+
+    /**
+     * Custom string appended as URL param to ElasticSearch requests for reproducible search results ordering.
+     * If user is logged in, use their userID, otherwise return generated string.
+     *
+     * @property esPreferenceKey
+     * @return {String}
+     */
+    esPreferenceKey: Ember.computed('currentUserId', function() {
```

Comment:
I agree with Abram's comment that this can be named more generically. Especially if you use a more proper hash I think it is safe to call this `sessionKey` and adjust the comment to say it is just a simple hash of the username that can identify the current session and that it is used for "... reproducible search results ordering..." etc.